pkg/handler/repository_rpms_test.go

```diff
@@ -0,0 +1,330 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/config"
+	"github.com/content-services/content-sources-backend/pkg/db"
+	test_handler "github.com/content-services/content-sources-backend/pkg/test/handler"
+	mock_dao "github.com/content-services/content-sources-backend/pkg/test/mocks"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/openlyinc/pointy"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gorm.io/gorm"
+)
+
+func serveRpmsRouter(req *http.Request, mockDao *mock_dao.RpmDao) (int, []byte, error) {
+	var (
+		err error
+	)
+
+	router := echo.New()
+	router.Use(middleware.RequestIDWithConfig(middleware.RequestIDConfig{
+		TargetHeader: "x-rh-insights-request-id",
+	}))
+	router.Use(config.WrapMiddlewareWithSkipper(identity.EnforceIdentity, config.SkipLiveness))
+	pathPrefix := router.Group(fullRootPath())
+
+	router.HTTPErrorHandler = config.CustomHTTPErrorHandler
+
+	rh := RepositoryRpmHandler{
+		Dao: mockDao,
+	}
+	RegisterRepositoryRpmRoutes(pathPrefix, &rh.Dao)
+
+	rr := httptest.NewRecorder()
+	router.ServeHTTP(rr, req)
+
+	response := rr.Result()
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	return response.StatusCode, body, err
+}
+
+type RpmSuite struct {
+	suite.Suite
+	savedDB *gorm.DB
+	echo    *echo.Echo
+}
+
+func (suite *RpmSuite) SetupTest() {
+	suite.savedDB = db.DB
+	db.DB = db.DB.Begin()
+	suite.echo = echo.New()
+	suite.echo.Use(middleware.RequestIDWithConfig(middleware.RequestIDConfig{
+		TargetHeader: "x-rh-insights-request-id",
+	}))
+	suite.echo.Use(config.WrapMiddlewareWithSkipper(identity.EnforceIdentity, config.SkipLiveness))
+}
+
+func (suite *RpmSuite) TearDownTest() {
+	//Rollback and reset db.DB
+	db.DB.Rollback()
+	db.DB = suite.savedDB
+	require.NoError(suite.T(), suite.echo.Shutdown(context.Background()))
+}
+
+func (suite *RpmSuite) TestRegisterRepositoryRpmRoutes() {
```

Comment:
is this already tested implicitly by `serveRpmsRouter`? I see that function calls `RegisterRepositoryRpmRoutes(pathPrefix, &rh.Dao)`

Reply:
thanks mate! you are right that both are covering the code!

I used to add one function/method, one test for it; `serveRpmsRouter` is a helper to make the unit tests simpler; but the responsibility for `serveRpmsRouter` is not to cover `RegistryRepositoryRpmsRoutes`, even if it is covered indirectly; in this case it is true that it does not make any difference, but it allows to grow the unit test independent from `serveRpmsRouter` if more complex behavior is added to `RegistryRepositoryRpmsRoute`, and keep the responsibility for each function separated (helper and unit test).

I would leave it as it is because of the above vision, but if I am overlooking something, please let me know; I am human, I do mistakes (many), and I learn from them.

Reply:
I guess my instinct would be to add a unit test for `RegistryRepositoryRpmsRoutes` if/when it becomes more complex. But I understand what you're saying, I'm okay leaving it like this.